From f8e5fee15f99fe093a585e6b9e0aeccfd63ac190 Mon Sep 17 00:00:00 2001
From: mytja <mitja@severkar.eu>
Date: Fri, 15 Nov 2024 20:51:41 +0100
Subject: [PATCH 3/5] Move vendor refresh before authenticator registrations

Fixes fingerprint on Samsung Galaxy A53 (Samsung HAL).

This was present in the "Bunch of FOD stuff -- commonize refreshing the
 services, start supporting AIDL Samsung" patch, however in A15, that
line has been mistakenly moved below mInjector.publishBinderService().

Change-Id: Ib55b35ecf81991dcdd3236bb96bcd6db4a33718f
---
 .../core/java/com/android/server/biometrics/AuthService.java   | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/biometrics/AuthService.java b/services/core/java/com/android/server/biometrics/AuthService.java
index 9ac961816..a2ea1f0e8 100644
--- a/services/core/java/com/android/server/biometrics/AuthService.java
+++ b/services/core/java/com/android/server/biometrics/AuthService.java
@@ -899,9 +899,10 @@ public class AuthService extends SystemService {
             hidlConfigs = null;
         }
 
+        refreshVendorServices();
         registerAuthenticators();
         mInjector.publishBinderService(this, mImpl);
-        refreshVendorServices();
+
         //samsungSysinputCommand("fod_icon_visible,1");
         if(samsungHasCmd("fod_enable") && (mSamsungFingerprint != null || mSamsungFingerprintAidl != null)) {
             samsungCmd("fod_enable,1,1,0");
-- 
2.34.1

